trillian/integration/hammer.go

```diff
@@ -398,33 +480,106 @@ func (s *hammerState) addChain(ctx context.Context) error {
 	}
 	submitted.leafHash = sha256.Sum256(append([]byte{ct.TreeLeafPrefix}, submitted.leafData...))
 	s.pending.tryAppendCert(time.Now(), s.cfg.MMD, &submitted)
-	glog.V(3).Infof("%s: Uploaded cert has leaf-hash %x", s.cfg.LogCfg.Prefix, submitted.leafHash)
+	glog.V(3).Infof("%s: Uploaded %s cert has leaf-hash %x", s.cfg.LogCfg.Prefix, choice, submitted.leafHash)
 	return nil
 }
 
 func (s *hammerState) addChainInvalid(ctx context.Context) error {
-	// Invalid because it's a pre-cert chain, not a cert chain.
-	chain, _, err := makePrecertChain(s.cfg.LeafChain, s.cfg.CACert, s.cfg.Signer, s.notAfter)
-	if err != nil {
-		return fmt.Errorf("failed to make fresh cert: %v", err)
+	choices := []Choice{EmptyChain, PrecertNotCert, NoChainToRoot, UnparsableCert, SignatureNotCorrect}
+	choice := choices[rand.Intn(len(choices))]
+
+	var err error
+	var chain []ct.ASN1Cert
+	switch choice {
+	case EmptyChain:
+	case PrecertNotCert:
+		chain, _, err = makePrecertChain(s.cfg.LeafChain, s.cfg.CACert, s.cfg.Signer, s.notAfter)
+		if err != nil {
+			return fmt.Errorf("failed to make chain(%s): %v", choice, err)
+		}
+	case NoChainToRoot:
+		chain, err = makeCertChain(s.cfg.LeafChain, s.cfg.LeafCert, s.cfg.CACert, s.cfg.Signer, s.notAfter)
+		if err != nil {
+			return fmt.Errorf("failed to make chain(%s): %v", choice, err)
+		}
+		chain = append(chain[:0], chain[2:]...)
```

Comment:
As discussed in person, to skip just the intermediate this should be 

chain = append(chain[:1], chain[2:]...)

because the slice[a:b] syntax returns a closed-open range :)

Reply:
Doh, fixed.